src/System.Management.Automation/engine/pipeline.cs

```diff
@@ -1298,7 +1298,16 @@ private void DisposeCommands()
                         // pipeline failure and continue disposing cmdlets.
                         try
                         {
-                            commandProcessor.CommandRuntime.RemoveVariableListsInPipe();
+                            var commandRuntime = commandProcessor.CommandRuntime;
+                            if (commandProcessor is CommandProcessor && commandProcessor.Command is not PSScriptCmdlet)
+                            {
+                                // Only a cmdlet can have variable lists defined via the common parameters.
+                                // We only need to take care of binary cmdlet here, because for script cmdlet,
+                                // the variable lists were already removed when exiting a scope.
+                                commandRuntime.RemoveVariableListsInPipe();
+                            }
```

Comment:
Is saving the variable remove operation going to be worth the reflection here? Just want to ask the question to confirm